### PR TITLE
[GEOS-7651] Use IOUtils rename method for FileSystemResource

### PR DESCRIPTION
**Pull request by tbarsballe**
Depends upon https://github.com/geoserver/geoserver/pull/1811; Travis build will fail until that PR is merged to master.

### CODE REVIEW

**Comment by aaime** [comment]
Uh, we have a compile failure here

**Comment by aaime** [comment]
Aaah ok, depends on the other one

**Comment by tbarsballe** [comment]
Now that the other one is merged, I have restarted the build.

File: src/restconfig/src/test/java/org/geoserver/rest/resources/ResourceTest.java [hunk 1]
```diff
@@ -295,7 +295,6 @@ public void testMove() throws Exception {
     
     @Test
     public void testMoveDirectory() throws Exception {
-        put("/rest/resource/mydir/mynewdir?operation=move", "/mydir");
```
Comment (tbarsballe) [inline src/restconfig/src/test/java/org/geoserver/rest/resources/ResourceTest.java]: As far as I can tell, this was intended to fail before, but works fine with the new Apache Commons FileUtils move/rename methods, causing the subsequent line to fail.

Hoping to get more details from @NielsCharlier

**Comment by tbarsballe** [comment]
Got some feedback from @NielsCharlier on the mailing lists, now updated to explicitly disallow renaming a resource to a descendant of itself.

File: src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java [hunk 2]
```diff
@@ -476,15 +478,24 @@ public boolean delete() {
...
-                return Resources.renameByCopy(this, dest);
+            if (dest.parent().path().contains(path())) {
+                LOGGER.log(Level.FINE, "Cannot rename a resource to a descendant of itself");
+                return false;
             }
+            try {
+                if(dest instanceof FileSystemResource) {
+                    rename(file, ((FileSystemResource)dest).file);
+                } else if(dest instanceof Files.ResourceAdaptor) {
+                    rename(file, ((Files.ResourceAdaptor)dest).file);
```
Comment (jodygarnett) [inline src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java]: Interesting we have to special case the adaptor, I am not sure how often these are used/useful.

**Comment by jodygarnett** [comment]
This looks good, all tests pass.

**Pull request merged**